stubs/tifftools/tifftools.pyi

```diff
@@ -0,0 +1,41 @@
+from os import PathLike
+from typing import BinaryIO, Literal, NotRequired, TypedDict
+
+_PathOrStream = str | PathLike[str] | BinaryIO
+
+class TiffInfo(TypedDict):
+    ifds: list[IFD]
+    path_or_fobj: _PathOrStream
+    size: int
+    header: bytes
+    bigEndian: bool
+    bigtiff: bool
+    endianPack: Literal['>', '<']
+    firstifd: int
+
+class IFD(TypedDict):
+    offset: int
+    tags: dict[int, TagEntry]
+    path_or_fobj: _PathOrStream
+    size: int
+    bigEndian: bool
+    bigtiff: bool
+    tagcount: int
+
+class TagEntry(TypedDict):
+    datatype: int
+    count: int
+    datapos: int
+    offset: NotRequired[int]
```

Comment:
Having an optional item in a `TypedDict` requires Python 3.11 to properly annotate: https://docs.python.org/3/whatsnew/3.11.html#pep-655-marking-individual-typeddict-items-as-required-or-not-required

This forces us to bump to Python 3.11 for the whole project, but I think having correct stubs is a strong enough reason to do so. I expected that we'd likely upgrade at some point regardless.

Reply:
I don't think requiring Python 3.11 is a good idea *yet*.  Some of our tile sources in large_image don't yet have wheels published for 3.11 (e.g., some of the dependencies for nd2 files and the tifffile library).  And, our collaborators probably haven't upgraded to 3.11 anywhere yet, either, so it will add unnecessary complication to their use of it.

Reply:
What about using [totality](https://mypy.readthedocs.io/en/latest/more_types.html#totality)? Maybe this doesn't solve the problem completely but it seems like a compromise between staying at 3.10 and having helpful type annotations.

Reply:
See #16. I'll change this PR now to use `total`.